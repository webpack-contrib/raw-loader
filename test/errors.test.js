import loader from '../index';

it('validation', () => {
  const validate = (options) =>
    loader.call(
      Object.assign(
        {},
        {
          query: options,
        }
      ),
      'Text'
    );

  expect(() => validate({})).not.toThrow();
  expect(() => validate(true)).not.toThrow();
  expect(() => validate({ unknown: true })).toThrowErrorMatchingSnapshot();
});
