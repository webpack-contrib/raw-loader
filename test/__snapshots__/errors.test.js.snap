// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation 1`] = `
"Raw Loader Invalid Options

options should NOT have additional properties
"
`;
